pkg/allocator/allocator.go

```diff
@@ -443,7 +443,7 @@ func (a *Allocator) lockedAllocate(ctx context.Context, key AllocatorKey) (idpoo
 		return 0, false, err
 	}
 
-	defer lock.Unlock(ctx)
+	defer lock.Unlock(context.TODO())
```

Comment:
I think we want `Background()` as this is not a context that has yet to be defined but it's a context that should never expire.

Reply:
ah that's a good point. Fixed